src/utils/onScroll/onScroll.js

```diff
@@ -0,0 +1,17 @@
+import {
+  isScrolledToLeft,
+  isScrolledToTop,
+  isScrolledToRight,
+  isScrolledToBottom,
+} from "../isScrolledTo";
+
+const onScroll = setState => ($container, $scroll) => {
```

Comment:
I think this function is doing two things and could do only one and be more reusable. Rename `onScroll` to maybe `getScrolledBounds` or something other that's more explicit. Then we can get rid of this first curried param `setState`. Also `onScroll` does not really tell what I would like this function to be used for it just tells when it should be call.